addons/web/static/src/views/relational_model/record.js

```diff
@@ -373,6 +374,7 @@ export class Record extends DataPoint {
             onChange: ({ withoutOnchange } = {}) =>
                 this._update({ [fieldName]: staticList }, { withoutOnchange }),
             parent: this,
+            relationField: this.fields[fieldName].relation_field,
```

Comment:
This would probably make sense in the config (alongside resModel and fields), but with a default to null or false as it's obviously optional in Record